src/reducers/versions.tsx

```diff
@@ -428,6 +431,64 @@ export const fetchVersionsList = ({
   };
 };
 
+type CreateInternalDiffParams = {
+  version: ExternalVersionWithDiff;
+  baseVersionId: number;
+  headVersionId: number;
+};
+
+// This function returns an array of objects compatible with `react-diff-view`.
+export const createInternalDiffs = ({
+  baseVersionId,
+  headVersionId,
+  version,
+}: CreateInternalDiffParams) => {
+  const GIT_STATUS_TO_TYPE: { [mode: string]: string } = {
+    A: 'add',
+    C: 'copy',
+    D: 'delete',
+    M: 'modify',
+    R: 'rename',
+  };
+
+  return version.file.diff.map(
+    (diff: ExternalDiff): DiffInfo => {
+      return {
+        newRevision: String(headVersionId),
+        oldRevision: String(baseVersionId),
+        hunks: diff.hunks.map((hunk: ExternalHunk) => ({
+          changes: hunk.changes.map((change: ExternalChange) => ({
```

Comment:
Observation: Sometimes we extract functions for these, such as `createInternalChange` and `createInternalHunk`, but I guess if you don't need to call those anywhere else (we often need them in tests), then there's no reason to extract them into their own functions.

Reply:
A `hunk` or a `change` per se does not have much value.